numba/tests/test_debuginfo.py

```diff
@@ -78,6 +78,25 @@ def _get_metadata(self, fn, sig):
                 metadata.append(line)
         return metadata
 
+    def _subprocess_test_runner(self, test_name):
+        # Runs a named unit test, test_name (str), from this class in a
+        # subprocess with `NUMBA_OPT` set to 0 so as to get the most unoptimized
+        # llvm IR available. It then checks the test executed without error.
+        themod = self.__module__
+        thecls = type(self).__name__
+        injected_method = f'{themod}.{thecls}.{test_name}'
+        cmd = [sys.executable, '-m', 'numba.runtests', injected_method]
+        env_copy = os.environ.copy()
+        env_copy['SUBPROC_TEST'] = '1'
+        env_copy['NUMBA_OPT'] = '0'
+        status = subprocess.run(cmd, stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE, timeout=60,
+                                env=env_copy, universal_newlines=True)
+        self.assertEqual(status.returncode, 0)
+        self.assertIn('OK', status.stderr)
+        self.assertTrue('FAIL' not in status.stderr)
+        self.assertTrue('ERROR' not in status.stderr)
```

Comment:
a duplication of https://github.com/numba/numba/blob/0c499bfff7ebe4fe5d8a6c1d20653e69f1f2a639/numba/tests/test_parfors.py#L73-L86

we should dedup here or in a immediate follow up PR